>From ee5c21ac6930de0cfdae2a71a4799524d8c33db0 Mon Sep 17 00:00:00 2001
From: Brett Smith <brettcsmith@brettcsmith.org>
Date: Wed, 15 Feb 2017 14:39:41 -0500
Subject: [PATCH] comparators.json: Catch bad JSON errors on Python pre-3.5.

json.JSONDecodeError was introduced in Python 3.5.  Earlier versions raised
ValueError, which is the base class of JSONDecodeError.  Catch that instead.
---
 diffoscope/comparators/json.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/diffoscope/comparators/json.py b/diffoscope/comparators/json.py
index d46c830..96e8002 100644
--- a/diffoscope/comparators/json.py
+++ b/diffoscope/comparators/json.py
@@ -25,7 +25,6 @@ from diffoscope.difference import Difference
 
 from .utils.file import File
 
-
 class JSONFile(File):
     RE_FILE_EXTENSION = re.compile(r'\.json$')
 
@@ -37,7 +36,7 @@ class JSONFile(File):
         with open(file.path) as f:
             try:
                 file.parsed = json.load(f, object_pairs_hook=collections.OrderedDict)
-            except json.JSONDecodeError:
+            except ValueError:
                 return False
 
         return True
-- 
2.1.4

